src/go/k8s/internal/controller/redpanda/user_controller.go

```diff
@@ -0,0 +1,229 @@
+			return handleErrors(err)
+		}
+		hasManagedUser = true
+	}
+
+	if hasUser && !shouldManageUser {
+		if err := usersClient.Delete(ctx, user); err != nil {
+			return handleErrors(err)
+		}
+		hasManagedUser = false
+	}
+
+	if shouldManageACLs {
+		if err := syncer.Sync(ctx, user); err != nil {
+			return handleErrors(err)
+		}
+		hasManagedACLs = true
+	}
+
+	if !shouldManageACLs && hasManagedACLs {
+		if err := syncer.DeleteAll(ctx, user); err != nil {
+			return handleErrors(err)
+		}
+		hasManagedACLs = false
+	}
+
+	return redpandav1alpha2.UserSyncedCondition(user.Name), hasManagedUser, hasManagedACLs, nil
+}
+
+func (r *UserReconciler) deleteUser(ctx context.Context, user *redpandav1alpha2.User) error {
+	hasManagedACLs, hasManagedUser := user.HasManagedACLs(), user.HasManagedUser()
+
+	ignoreAllConnectionErrors := func(err error) error {
+		// If we have known errors where we're unable to actually establish
+		// a connection to the cluster due to say, invalid connection parameters
+		// we're going to just skip the cleanup phase since we likely won't be
+		// able to clean ourselves up anyway.
+		if internalclient.IsTerminalClientError(err) ||
+			internalclient.IsConfigurationError(err) ||
+			internalclient.IsInvalidClusterError(err) {
+			return nil
+		}
+		return err
+	}
+
+	usersClient, syncer, hasUser, err := r.userAndACLClients(ctx, user)
+	if err != nil {
+		return ignoreAllConnectionErrors(err)
+	}
+
+	if hasUser && hasManagedUser {
+		if err := usersClient.Delete(ctx, user); err != nil {
+			return ignoreAllConnectionErrors(err)
+		}
+	}
+
+	if hasManagedACLs {
+		if err := syncer.DeleteAll(ctx, user); err != nil {
+			return ignoreAllConnectionErrors(err)
+		}
+	}
+
+	return nil
+}
+
+func (r *UserReconciler) userAndACLClients(ctx context.Context, user *redpandav1alpha2.User) (*users.Client, *acls.Syncer, bool, error) {
+	usersClient, err := r.Users(ctx, user, r.extraOptions...)
+	if err != nil {
+		return nil, nil, false, err
+	}
+
+	syncer, err := r.ACLs(ctx, user, r.extraOptions...)
+	if err != nil {
+		return nil, nil, false, err
+	}
+
+	hasUser, err := usersClient.Has(ctx, user)
+	if err != nil {
+		return nil, nil, false, err
+	}
+
+	return usersClient, syncer, hasUser, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *UserReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager) error {
+	if err := registerUserClusterIndex(ctx, mgr); err != nil {
+		return err
+	}
+
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&redpandav1alpha2.User{}).
+		Owns(&corev1.Secret{}).
+		Watches(&redpandav1alpha2.Redpanda{}, handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, o client.Object) []reconcile.Request {
+			requests, err := usersForCluster(ctx, r.KubernetesClient(), client.ObjectKeyFromObject(o))
+			if err != nil {
+				mgr.GetLogger().V(1).Info("skipping reconciliation due to fetching error", "error", err)
+				return nil
+			}
+			return requests
+		})).
```

Comment:
So, swallowing errors like this is pretty typical for issues in an `EnqueueRequestsFromMapFunc`. If we want to do something heavier, I could introduce some sort of internal requeueing mechanism similar to what I've played around with [here](https://github.com/andrewstucki/cr-enqueuer/blob/b262d8dc15ccc0295024870ed2ee405b71ec3190/internal/controller/enqueuer.go#L38).

That said, we should pretty much never run into this error as the `List` operation that it's doing under the hood is using a client-side cache index, meaning it's only ever going to read from cache and should never hit the network.